server.go

```diff
@@ -490,7 +490,7 @@ func (s *Server) shutdown() {
 			return false
 		}
 
-		if len(c.consumers) > 0 {
+		if c.consID != "" {
```

Comment:
Can't we reuse `fetchConsumer()` here? If so, we should make it exported (`FetchConsumer`).

Reply:
`Consumer` is the `fetchConsumer` alternative. However, is not straightforward to call that function without refactoring the code in this place.